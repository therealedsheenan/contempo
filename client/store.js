import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import { createEpicMiddleware, combineEpics } from 'redux-observable';
import 'rxjs';

// redux settings
import greetingReducer from '../src/redux/greeting/reducer';
import { greetingEpic } from '../src/redux/greeting/actions';

const combinedEpics = combineEpics(greetingEpic);

const epicMiddleWare = createEpicMiddleware(combinedEpics);

const store = createStore(
  combineReducers({
    greetingReducer,
    routing: routerReducer
  }),
  compose(
    applyMiddleware(epicMiddleWare),
    typeof window === 'object' &&
    typeof window.devToolsExtension !== 'undefined'
      ? window.devToolsExtension()
      : f => f
  )
);

export default store;
