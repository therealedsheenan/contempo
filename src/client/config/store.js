import { createStore, applyMiddleware, compose, combineReducers } from 'redux'
// import thunk from 'redux-thunk'
import { routerReducer } from 'react-router-redux'
import { createEpicMiddleware } from 'redux-observable'

// redux settings
import greetingReducer, { greetingEpic } from '../redux/modules/greeting'

const epicMiddleWare = createEpicMiddleware(greetingEpic)

export const store = createStore(
  combineReducers({
    greetingReducer,
    routing: routerReducer
  }),
  compose(
    applyMiddleware(epicMiddleWare),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : (f) => f
  )
)

