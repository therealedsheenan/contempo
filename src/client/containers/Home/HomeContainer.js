import React from 'react'
import { Link } from 'react-router'
import { GreetingComponent } from 'components'

const HomeContainer = React.createClass({
  render () {
    return (
      <div>
        <GreetingComponent />
        <Link to='/styleguide'>{'Styleguide'}</Link>
      </div>
    )
  }
})

export default HomeContainer
