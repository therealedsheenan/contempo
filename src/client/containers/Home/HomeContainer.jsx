import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { requestGreeting } from '../../redux/greeting/actions';

import GreetingComponent from '../../components/Greeting/GreetingComponent';

class HomeContainer extends React.Component {
  componentDidMount() {
    this.props.requestGreeting();
  }

  render() {
    return <GreetingComponent message={this.props.greeting.content} />;
  }
}

HomeContainer.propTypes = {
  requestGreeting: PropTypes.func.isRequired,
  greeting: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]).isRequired
};

const mapStateToProps = ({ greetingReducer }) => {
  const { fetching, greeting } = greetingReducer;

  return {
    fetching,
    greeting
  };
};

const mapDispatchToProps = dispatch => ({
  requestGreeting: () => dispatch(requestGreeting())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeContainer);
