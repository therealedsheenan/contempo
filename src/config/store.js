// @flow

import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';
import { createEpicMiddleware, combineEpics } from 'redux-observable';
// the only needed rx files for redux-observables
import 'rxjs/Observable';
import 'rxjs/add/operator/mergeMap';
import 'rxjs/add/operator/toPromise';

// redux settings
import greetingReducer from '../redux/greeting/reducer';
import { greetingEpic } from '../redux/greeting/actions';

const combinedEpics = combineEpics(greetingEpic);

const epicMiddleWare = createEpicMiddleware(combinedEpics);

const store = createStore(
  combineReducers({
    greetingReducer,
    routing: routerReducer
  }),
  compose(
    applyMiddleware(epicMiddleWare),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
  )
);

export default store;
