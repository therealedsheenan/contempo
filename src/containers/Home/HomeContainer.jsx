// @flow

import React from 'react';
import { connect } from 'react-redux';
import { requestGreeting } from '../../redux/greeting/actions';
import GreetingComponent from '../../components/Greeting/GreetingComponent';

class HomeContainer extends React.Component {
  componentWillMount() {
    this.props.requestGreeting();
  }

  props: {
    requestGreeting: Function,
    greeting: Object
  };

  render() {
    return <GreetingComponent message={this.props.greeting.content} />;
  }
}

const mapStateToProps = ({ greetingReducer }) => {
  const { fetching, greeting } = greetingReducer;

  return {
    fetching,
    greeting
  };
};

const mapDispatchToProps = dispatch => ({
  requestGreeting: () => dispatch(requestGreeting())
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer);
