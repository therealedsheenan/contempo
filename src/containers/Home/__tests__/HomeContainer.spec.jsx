import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import HomeContainer from '../HomeContainer';
import store from '../../../config/store';

test('Home existence', () => {
  const component = renderer.create(
    <Provider store={store}>
      <HomeContainer />
    </Provider>
  ).toJSON();

  // correct home component
  expect(component).toMatchSnapshot();
});
