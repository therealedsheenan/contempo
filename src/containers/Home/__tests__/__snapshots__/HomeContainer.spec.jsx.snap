// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home existence 1`] = `
<div
  className="styles__Wrapper-dhcWqx cQtooR"
>
  <h1
    className="styles__Title-kJmJYc kAsCYU"
  >
    Good 
    Day!
  </h1>
  <p
    className="styles__SubTitle-geuJrq dzWtlk"
  >
    Welcome to
  </p>
  <div>
    <img
      alt=""
      src="./assets/images/contempo.png"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
